neptune/new/internal/backends/utils.py

```diff
@@ -55,7 +65,7 @@ def wrapper(*args, **kwargs):
                     requests.exceptions.ConnectionError, requests.exceptions.Timeout,
                     HTTPRequestTimeout, HTTPServiceUnavailable, HTTPGatewayTimeout, HTTPBadGateway,
                     HTTPTooManyRequests, HTTPServerError) as e:
-                time.sleep(2 ** retry)
+                time.sleep(min(2 ** retry, MAX_RETRY_TIME))
```

Comment:
Not sure if you want to always compute `2 ** retry` -- this can be a BIG number...